libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2222,24 +2247,20 @@ string YulUtilFunctions::readFromStorageReferenceType(Type const& _type)
 	for (size_t i = 0; i < structMembers.size(); ++i)
 	{
 		auto const& [memberSlotDiff, memberStorageOffset] = structType.storageOffsetsOfMember(structMembers[i].name);
+		solAssert(structMembers[i].type->isValueType() || memberStorageOffset == 0, "");
 
 		memberSetValues[i]["setMember"] = Whiskers(R"(
 			{
-				let <memberValues> := <readFromStorage>(add(slot, <memberSlotDiff>)<?hasOffset>, <memberStorageOffset></hasOffset>)
+				let <memberValues> := <readFromStorage>(add(slot, <memberSlotDiff>), <memberStorageOffset>)
 				<writeToMemory>(add(value, <memberMemoryOffset>), <memberValues>)
```

Comment:
Thinking about this. 
What happens is that `readFromStorage` copies from storage to memory in newly allocated space.
Than `writeToMemory` copies it to another location in memory.

Instead of doing this, I think it will be better to do direct copy to the desired location in memory to remove unnecessary copies.

I can do it in a separate PR, as this wasn't really scope of this one.

Reply:
Not true. For value types it is fine, and for reference it will just store reference, so we can ignore my concern here :-)